docs/design/downward_api_resources_limits_requests.md

```diff
@@ -0,0 +1,537 @@
+     resources:
+        requests:
+          memory: "64Mi"
+	  cpu: "250m"
+        limits:
+          memory: "128Mi"
+          cpu: "500m"
+      volumeMounts:
+        - name: podinfo
+          mountPath: /etc
+          readOnly: false
+  volumes:
+    - name: podinfo
+      downwardAPI:
+        items:
+          - path: "cpu_limit"
+            containerFieldRef:
+              containerName: "client-container"
+              fieldPath: resources.limits.cpu
+```
+#### Validations
+
+For APIs with partial json path selectors, verify
+that selectors are valid relative to container spec.
+Also verify that container name is provided with volumes.
+
+
+### API with no selectors
+
+In this approach, users specify fixed strings to retrieve resources
+limits and requests. This approach is similar to the existing downward
+API implementation approach. These will be implemented by introducing a
+`ResourceFieldSelector` (json: `resourceFieldRef`) to extend the current
+implementation for `type DownwardAPIVolumeFile struct` and `type EnvVarSource struct`:
+
+```
+type ResourceFieldSelector struct {
+        // Container name
+        Name string `json:"containerName,omitempty"`
+        // Required: Resource to select
+        Resource string `json:"resource"`
+}
+
+// Represents a single file containing information from the downward API
+type DownwardAPIVolumeFile struct {
+     // Required: Path is  the relative path name of the file to be created.
+     Path string `json:"path"`
+     // Selects a field of the pod: only annotations, labels, name and
+     // namespace are supported.
+     FieldRef *ObjectFieldSelector `json:"fieldRef, omitempty"`
+     // Selects a resource of the container: only resources limits and requests
+     // (cpu, memory) are currently supported.
+     ResourceFieldRef *ResourceFieldSelector `json:"resourceFieldRef,omitempty"`
+}
+
+// EnvVarSource represents a source for the value of an EnvVar.
+// Only one of its fields may be set.
+type EnvVarSource struct {
+   // Required: Selects a resource of the container: only resources limits and
+   // requests (cpu, memory) are supported.
+   ResourceFieldRef *ResourceFieldSelector `json:"resourceFieldRef,omitempty"`
+   // Selects a field of the pod; only name and namespace are supported.
+   FieldRef *ObjectFieldSelector `json:"fieldRef,omitempty"`
+   // Selects a key of a ConfigMap.
+   ConfigMapKeyRef *ConfigMapKeySelector `json:"configMapKeyRef,omitempty"`
+   // Selects a key of a secret in the pod's namespace.
+   SecretKeyRef *SecretKeySelector `json:"secretKeyRef,omitempty"`
+}
+```
+
+#### Environment variables
+
+This table shows how selectors can be used for various requests and
+limits to be exposed as environment variables.
+
+| Name | Resource |
+| -------------------- | -------------------|
+| CPU_LIMIT | cpu_limit |
```

Comment:
I would rather use a syntax for magic keys that would also be a valid selector in the future, basically the same selector as for other approaches, just implemented as a magic key.